src/hooks/useHailFrequency.js

```diff
@@ -25,38 +26,37 @@ async function fetchFromHailFrequency({
   }).then((res) => res.json())
 }
 
-async function fetcher(query, spaceId, spaceToken) {
+async function fetcher(query, spaceId, spaceToken, endpoint, type) {
   let data = []
   let nextToken = ''
-  let page = 1
-
-  while (page === 1 || nextToken) {
+  // fetch the data as long as there's more
+  // use a do_while loop since we need to fetch at least 1 time
+  do {
```

Comment:
Used a do.. while here because we need to run the loop at least 1 time. Mostly commenting because it's not as common of a pattern in JS.